presto-main/src/test/java/io/prestosql/sql/planner/TestLogicalPlanner.java

```diff
@@ -250,6 +250,28 @@ public void testAllFieldsDereferenceFromNonDeterministic()
                                         project(
                                                 ImmutableMap.of("rand", expression("rand()")),
                                                 values())))));
+
+        assertPlan("SELECT (rand(), rand()).* FROM (VALUES 1) t(x)",
+                any(
+                        project(
+                                ImmutableMap.of(
+                                        "output_1", expression("CAST(r AS ROW(f0 double,f1 double)).f0"),
+                                        "output_2", expression("CAST(r AS ROW(f0 double,f1 double)).f1")),
+                                project(
+                                        ImmutableMap.of("r", expression("ROW(rand(), rand())")),
+                                        values()))));
+
+        assertPlan("SELECT (rand(), rand()).* FROM (VALUES 1, 2) t(x) ORDER BY 1",
```

Comment:
what does this case test specifically vs previous case? Consider adding a comment

Reply:
It's to check that the ORDER BY clause doesn't "duplicate" the expression, which would result in incorrect results due to non-determinism.